src/Check/Resource/Http/StatusCheck.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Leankoala\HealthFoundation\Check\Resource\Http;
+
+use Leankoala\HealthFoundation\Check\Check;
+use Leankoala\HealthFoundation\Check\Result;
+
+class StatusCheck implements Check
+{
+    const IDENTIFIER = 'base:resource:http:status';
+
+    private $destination = null;
+    private $httpStatus = null;
+
+    public function init($destination,$httpStatus)
+    {
+        $this->destination = $destination;
+		$this->httpStatus = $httpStatus;
+    }
+
+    /**
+     * Checks the response status 
+     *
+     * @return Result
+     */
+    public function run()
+    {
+		$ch = curl_init();
+
+		curl_setopt( $ch, CURLOPT_URL, $this->destination );
```

Comment:
How about using Guzzle? 
http://docs.guzzlephp.org/en/stable/

Had the idea to just inject a guzzle client into the class from within the healthFoundation class.

Reply:
I added an interface ```HttpClientAwareCheck```. If a check implements it it automatically gets the GuzzleClient injected. 

See here: https://github.com/Leankoala/HealthFoundation/blob/master/src/HealthFoundation.php

Reply:
? Why HttpClientAwareCheck is an interface with "setHttpClient"? Then i have to implement the httpClient manually in my check? My check has no connection to the HealthFoundation class, so i can't get guzzle via getHttpClient. I did not understand that move right now :-) Can you give a little hint/example for that?